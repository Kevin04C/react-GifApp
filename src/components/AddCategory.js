import React, { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({ setcategories }) => {
  const [inputValue, setInputValue] = useState("");

  const handleInputChange = (e) => {
    setInputValue(e.target.value);

    console.log("handleInputChange llamado",inputValue);
  };

  const handleSubmit = (e) => {
    e.preventDefault();

    if (inputValue.trim().length > 2) {
      setcategories((categories) => [inputValue, ...categories]);
      setInputValue("");
    } else {
      alert("No podes dejar el campo vacio");
    }
  };

  return (
    <form onSubmit={handleSubmit}>
      <p>{inputValue}</p>
      <input
        type="text"
        value={inputValue}
        onChange={handleInputChange}
        id="input"
        autoComplete="off"
      />
    </form>
  );
};

AddCategory.propTypes = {
  setcategories: PropTypes.func.isRequired,
};
