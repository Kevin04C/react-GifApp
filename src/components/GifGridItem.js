import React from 'react';

export const GifGridItem = ({id,title, url}) => {

  return (
    <div className='gifs__item animate__animated animate__fadeInUpBig '>
      <img src={url} alt={title}/>
      <p> { title }</p>
    </div>
  )
}
