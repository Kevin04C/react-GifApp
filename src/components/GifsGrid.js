import React from "react";
import PropTypes from 'prop-types';
import { GifGridItem } from "./GifGridItem";
import laoder from '../assets/loader.svg'
import { useFetchGifs } from "../hooks/useFetchGifs";

export const GifsGrid = ({ category }) => {
  const { data: Images, laoding } = useFetchGifs( category );

  console.log(Images);

  return (
    <>
      <h3>{category.toUpperCase()}</h3>
      
      {laoding && <img src={laoder} alt="cargando" className="loader" />}

      <div className="gifs__container">
        {Images.map((image) => (
          <GifGridItem {...image} key={image.id} />
        ))}
      </div>
    </>
  );
};

GifsGrid.propTypes = {
  category : PropTypes.string.isRequired,
}