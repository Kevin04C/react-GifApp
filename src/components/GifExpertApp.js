import React, { useState } from "react";
import { AddCategory } from "./AddCategory";
import { GifsGrid } from "./GifsGrid";

export const GifExpertApp = () => {
  const [categories, setcategories] = useState();
  return (
    <>
      <h2>GifExpertApp</h2>
      <AddCategory setcategories={setcategories} />
      <hr />
      {categories.map((category) => (
        <GifsGrid category={category} key={category}/>
      ))}
    </>
  );
};
