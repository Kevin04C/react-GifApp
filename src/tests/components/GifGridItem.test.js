import React from "react";
import { shallow } from "enzyme";
import { GifGridItem } from "../../components/GifGridItem";

describe("Prueba en el archivo GifGridItem", () => {
  const title = "TÃ­tulo de prueba";
  const url = "https://localhost.jpg";

  let wrapper = shallow(<GifGridItem url={url} title={title} />);

  test("prueba de snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe devolver el title correctamente", () => {
    const p = wrapper.find("p");

    expect(p.text().trim()).toBe(title);
  });

  test("debe devolver el src y el alt correctamente", () => {
    const img = wrapper.find("img");

    expect(img.props().src).toBe(url);
    expect(img.props().alt).toBe(title);
  });

  test("debe devolver la clase animate__fadeInUpBig", () => {
    const div = wrapper.find("div");

    expect(div.hasClass("animate__fadeInUpBig")).toEqual(true);

  });
});
