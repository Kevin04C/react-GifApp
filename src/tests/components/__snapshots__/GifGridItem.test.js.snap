// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en el archivo GifGridItem prueba de snapshot 1`] = `
<div
  className="gifs__item animate__animated animate__fadeInUpBig "
>
  <img
    alt="Título de prueba"
    src="https://localhost.jpg"
  />
  <p>
     
    Título de prueba
  </p>
</div>
`;
